### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by pyhys**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,12 +52,14 @@ Features:
 
 Limitations or not yet implemented:
 
+* This is a device stack, so the IO-controller / master / PLC side is not supported.
```
Comment (olbjo) [inline README.md:55]: May "so" shall be replaced with "this means"

File: doc/applications_and_porting.rst [hunk 2]
```diff
@@ -27,3 +27,73 @@ Required features of the hardware
...
+During conformance testing, the PLC watchdog timeout setting is 3 x
+frame-to-frame interval times, ie the PLC will trigger an alarm if three
+consecutive cyclic data frames are missing. Thus the smallest allowed
+frame-to-frame interval given in the GSDML file must take the smallest
+useful PLC watchdog timeout setting into account.
+
+There can be different factors limiting the ability to send Profinet frames at
+a regular pace:
+
+* The initial work at Profinet communication startup should not
```
Comment (olbjo) [inline doc/applications_and_porting.rst:50]: reword

File: doc/applications_and_porting.rst [hunk 3]
```diff
@@ -27,3 +27,73 @@ Required features of the hardware
...
+  of Profinet cyclic data
+* Other work done by the Profinet application, for example work done in
+  callbacks, might slow down the communication.
+* Operating system jitter, which basically is that the operating system is
+  busy doing other work, will cause frame delays
+* Using a network interfaces connected via USB can cause additional frame
+  delays.
+
+For real-time operating systems (RTOS) for example rt-kernel, the
+frame-to-frame transmission will have a nice periodicity. The amount of work
```
Comment (olbjo) [inline doc/applications_and_porting.rst:61]: "less jitter"   ? (nice periodicity)

File: doc/applications_and_porting.rst [hunk 4]
```diff
@@ -27,3 +27,73 @@ Required features of the hardware
...
+delay between the frames, then you need to use the next standard PLC timeout
+value which is 12 ms. In order to handle that with a watchdog setting of 3 x
+frame interval, you need to use a minimum cycle time of 4 ms (as
+will be stated in your GSDML file).
+
+
+Example measurement results
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Measurements with p-net on a XMC4800 board running rt-kernel shows that the
+hardware is able to send frames with a nice periodicity of 1 ms. The liming
```
Comment (olbjo) [inline doc/applications_and_porting.rst:91]: "nice periodicity" -> little jitter ?

File: doc/compliancetest.rst [hunk 5]
```diff
@@ -183,21 +187,27 @@ Connection of the switch ports is described in the table below:
 The Automated RT tester will detect "Device B" by itself. No configuration is
 required in the Automated RT tester menu.
 
+The switch can send its portID as ``port001.b`` instead of ``port001``, if the
```
Comment (olbjo) [inline doc/compliancetest.rst:190]: Maybe the port names used are "port-001"  ?

**Review by olbjo — APPROVED** [review]
Great! Some minor comments.

File: README.md (hunk 1 above)
Comment (pyhys) [reply README.md]: Done

File: doc/applications_and_porting.rst (hunk 2 above)
Comment (pyhys) [reply doc/applications_and_porting.rst]: Done

File: doc/applications_and_porting.rst (hunk 3 above)
Comment (pyhys) [reply doc/applications_and_porting.rst]: Done

File: doc/applications_and_porting.rst (hunk 4 above)
Comment (pyhys) [reply doc/applications_and_porting.rst]: Done

File: doc/compliancetest.rst (hunk 5 above)
Comment (pyhys) [reply doc/compliancetest.rst]: Done

File: doc/applications_and_porting.rst [hunk 6]
```diff
@@ -27,3 +27,75 @@ Required features of the hardware
 
 * It should be possible to send and receive raw (layer 2) Ethernet frames
 * It should be possible to store data between runs, in a file system or some nonvolatile memory
+
+
+Minimum cycle time for your application and hardware
+----------------------------------------------------
+At PLC configuration, there are two adjustable parameters regarding the
+cyclic data timing for an IO-device. One basically controls the
+frame-to-frame transmission interval (given as a mulitple of the cycle time),
```
Comment (hefloryd) [inline doc/applications_and_porting.rst:36]: mutiple

File: doc/applications_and_porting.rst [hunk 7]
```diff
@@ -27,3 +27,75 @@ Required features of the hardware
...
+
+* The initial work at Profinet communication startup should not cause
+  delays in the cyclic data transmission once started.
+* The hardware needs to keep up with the steady-state sending and receiving
+  of Profinet cyclic data.
+* Other work done by the Profinet application, for example work done in
+  callbacks, might slow down the communication.
+* Operating system jitter, which basically is that the operating system is
+  busy doing other work, will cause frame delays.
+* Using a network interfaces connected via USB can cause additional frame
```
Comment (hefloryd) [inline doc/applications_and_porting.rst:58]: _a network interface_ or _network interfaces_

File: doc/applications_and_porting.rst [hunk 8]
```diff
@@ -27,3 +27,75 @@ Required features of the hardware
...
+  delays.
+
+For real-time operating systems (RTOS) for example rt-kernel, the
+frame-to-frame transmission will have little jitter. The amount of work
+needs to be done during Profinet communication setup or frame transmission is
```
Comment (hefloryd) [inline doc/applications_and_porting.rst:63]: _that_ needs to be done?

What does this refer to? If there is a timing problem during startup then that may just be a problem with the current implementation.

File: doc/applications_and_porting.rst (hunk 6 above)
Comment (pyhys) [reply doc/applications_and_porting.rst:36]: Done

File: doc/applications_and_porting.rst (hunk 7 above)
Comment (pyhys) [reply doc/applications_and_porting.rst:58]: Done

File: doc/applications_and_porting.rst (hunk 8 above)
Comment (pyhys) [reply doc/applications_and_porting.rst:63]: I removed "The amount of work ..." as it is not really adding any information.
Done.

**Pull request merged**